core/src/main/java/io/questdb/std/datetime/DateLocaleFactory.java

```diff
@@ -52,6 +53,16 @@ public static void load() {
     }
 
     public DateLocale getLocale(CharSequence id) {
-        return dateLocales.get(id);
+        DateLocale dateLocale = dateLocales.get(id);
+        if (dateLocale == null) {
+            return null;
+        }
+        if (dateLocale != dummyLocale) {
+            return dateLocale;
+        }
+        return dateLocales.compute(id, (key, val) -> {
+            Locale locale = Locale.forLanguageTag(id.toString());
```

Comment:
Currently it's a capturing lambda, so there will be an extra allocation per invocation:
```suggestion
            Locale locale = Locale.forLanguageTag(key.toString());
```

P.S. We tend to extract lambdas into method references to make sure that they're non-capturing.